### PR TITLE
FC-0001: Remove EdxRestApiClient usage in edx-analytics-dashboard

### PR DESCRIPTION
**Pull request by UvgenGen**
JIRA: [DEPR-131](https://openedx.atlassian.net/browse/DEPR-131)
https://github.com/openedx/public-engineering/issues/45

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @UvgenGen! I've created [OSPR-6586](https://openedx.atlassian.net/browse/OSPR-6586) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Review by dianakhuang — APPROVED** [review]
This looks good to me. Will pass along the PR to the owning team.

File: common/clients.py [hunk 1]
```diff
@@ -27,19 +26,30 @@ def all_courses(self):
...
+                        'page': page,
+                        'page_size': 100
+                    }
+                ).json()
                 course_details = response['results']
 
                 courses += course_details
 
-                if response['next']:
+                if response['pagination']['next']:
```
Comment (ashultz0) [inline common/clients.py:40]: this change seems unrelated to the client change, what motivated it?

Comment (UvgenGen) [reply common/clients.py:40]: Yes, this change is not related to the client change. As I see the response from the courses api has been changed, and the `next` is now contained in the `pagination`. 
Otherwise I get a KeyError.

**Review by ashultz0 — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@UvgenGen 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-analytics-dashboard&entry.867055334=https://github.com/openedx/edx-analytics-dashboard/pull/1282&entry.1484655318=https://github.com/UvgenGen&entry.752974735=2022-04-07+11:03&entry.1917517419=2022-04-08+14:10&entry.2133058324=Yes) so we can improve your experience in the future.
